### PR TITLE
added type mapping and support for different types

### PR DESCRIPTION
**Pull request by spuetz**
As there has been no reading and writing conversion methods for the PointCloud2 buffer, I implemented these methods. Also the new convertPointCloud2ToPointCloud supports different data types.

### CODE REVIEW

**Comment by spuetz** [comment]
As suggested by you, @vrabaud , I created a new pull request for indigo. It would be nice, if we can manage this pull request.

File: sensor_msgs/include/sensor_msgs/point_field_conversion.h [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  template<> struct typeAsPointFieldType<uint16_t> { static const uint8_t value = sensor_msgs::PointField::UINT16;  };
+  template<> struct typeAsPointFieldType<int32_t>  { static const uint8_t value = sensor_msgs::PointField::INT32;   };
+  template<> struct typeAsPointFieldType<uint32_t> { static const uint8_t value = sensor_msgs::PointField::UINT32;  };
+  template<> struct typeAsPointFieldType<float>    { static const uint8_t value = sensor_msgs::PointField::FLOAT32; };
+  template<> struct typeAsPointFieldType<double>   { static const uint8_t value = sensor_msgs::PointField::FLOAT64; };
+ 
+  /*!
+   * \Converts a value at the given pointer position, interpreted as the datatype
+   *  specified by the given template argument point_field_type, to the given
+   *  tampleate type T and returns it.
```
Comment (vrabaud) [inline sensor_msgs/include/sensor_msgs/point_field_conversion.h:85]: typo: should read "template"

File: sensor_msgs/include/sensor_msgs/point_field_conversion.h [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+   */
+  template<int point_field_type, typename T>
+    inline T readPointCloud2BufferValue(const unsigned char* data_ptr){
+      typedef typename pointFieldTypeAsType<point_field_type>::type type;
+      return static_cast<T>(*(reinterpret_cast<type const *>(data_ptr)));
+    }
+  
+  /*!
+   * \Converts a value at the given pointer position interpreted as the datatype
+   *  specified by the given datatype parameter to the given tampleate type and returns it.
```
Comment (vrabaud) [inline sensor_msgs/include/sensor_msgs/point_field_conversion.h:98]: typo: should read "template"

**Comment by vrabaud** [comment]
@spuetz , thx for fixing the typo and squashing
@tfoote , we can merge and release, lgtm.

**Comment by tfoote** [comment]
Merged with a rebase for type fixups: 26b33aa4b1018e5afad7bd235bdce3ed4fc58a85

This also fixes #81 and resolves the issues in #82 discussion.

**Comment by tfoote** [comment]
cherry-picked to jade fe7a5d226b47359cebf9dd0c567d6dbd18aba275

**Pull request closed without merging**